service/api_distro.go

```diff
@@ -29,3 +31,25 @@ func (as *APIServer) GetDistro(w http.ResponseWriter, r *http.Request) {
 	h.Distro.ProviderSettingsList = nil
 	gimlet.WriteJSON(w, h.Distro)
 }
+
+func (as *APIServer) GetDistroView(w http.ResponseWriter, r *http.Request) {
+	t := MustHaveTask(r)
+
+	h, err := host.FindOne(host.ByRunningTaskId(t.Id))
```

Comment:
I would add the checkHost middleware to the route handler, and get the host here the same way that checkHost does. That way we will have more consistent behavior in the rare scenario where a task ends up getting dispatched twice

Reply:
It would almost certainly be a bug/race if it happens, but because hosts can independently ask for their next_task and this request can get served by separate app servers, it's possible that 2 hosts get the same task. There are checks on the next_task queries to prevent this thankfully. Where I'm coming from is that because the host making the request identifies itself, that is theoretically a more reliable source than what the evergreen db thinks is running the task, though in practice it probably won't make a difference